github/githubapp.go

```diff
@@ -17,18 +19,38 @@ type GitHubApp struct {
 	ClientId       string `json:"client_id"`
 	InstallationId int    `json:"installation_id"`
 	PrivateKey     string `json:"private_key"` // SENSITIVE
+	parsedRSAKey   *rsa.PrivateKey
 }
 
-func (app *GitHubApp) IsZero() bool {
-	return *app == GitHubApp{}
-}
+// Validate validates the GitHubApp configuration. Returns an error if
+// GitHubApp is misconfigured.
+func (app *GitHubApp) Validate() error {
+	var mandatory []string
 
-// generateJWTtoken returns a signed JWT token used to authenticate as GitHub App
-func generateJWTtoken(clientId, privateKey string) (string, error) {
-	key, err := jwt.ParseRSAPrivateKeyFromPEM([]byte(privateKey))
+	if app.ClientId == "" {
+		mandatory = append(mandatory, "github_app.client_id")
+	}
+	if app.InstallationId == 0 {
+		mandatory = append(mandatory, "github_app.installation_id")
+	}
+	if app.PrivateKey == "" {
+		mandatory = append(mandatory, "github_app.private_key")
+	}
+
+	if len(mandatory) > 0 {
+		return fmt.Errorf("github_app: missing mandatory keys: %s", strings.Join(mandatory, ", "))
+	}
+
+	key, err := jwt.ParseRSAPrivateKeyFromPEM([]byte(app.PrivateKey))
 	if err != nil {
-		return "", fmt.Errorf("could not parse private key: %w", err)
+		return fmt.Errorf("github_app: could not parse private key: %w", err)
```

Comment:
Does the error you get from jwt tell you it should be in PEM format? If not, that would be worthwhile to mention it.

Reply:
This would be the full error (see the test below):

> github_app: could not parse private key: invalid key: Key must be a PEM encoded PKCS1 or PKCS8 ke